### PR TITLE
feat: Support mc find --count

### PR DESCRIPTION
**Pull request by jiuker**
## Description

feat: Support mc find --count
related [#17490](https://github.com/minio/minio/issues/17490)

## How to test this PR?

```shell
 ./mc find --newer-than 1d  minio       
minio/mytest/mytest.zip
minio/mytest/main.go
```

```shell
./mc find --newer-than 1d --count minio
2
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/find-main.go [hunk 1]
```diff
@@ -85,6 +85,10 @@ var (
 			Name:  "watch",
 			Usage: "monitor a specified path for newly created object(s)",
 		},
+		cli.BoolFlag{
+			Name:  "count",
+			Usage: "only print count numbers for matched object(s)",
```
Comment (shtripat) [inline cmd/find-main.go:90]: %s/only print count numbers for matched object(s)/only print numbers of matched object(s)/g

File: cmd/find-main.go [hunk 2]
```diff
@@ -171,6 +175,9 @@ EXAMPLES:
 
   11. Copy all versions of all objects in bucket in the local machine
       {{.Prompt}} {{.HelpName}} s3/bucket --versions --exec "mc cp --version-id {version} {} /tmp/dir/{}.{version}"
+
+  12. Count list all objects up to 3 levels sub-directory deep under "s3/bucket".
```
Comment (shtripat) [inline cmd/find-main.go:179]: %s/Count list all objects up to 3 levels sub-directory deep under "s3/bucket"/Count of objects up to 3 levels sub-directory deep under "s3/bucket"/g

File: cmd/find.go [hunk 3]
```diff
@@ -60,6 +61,27 @@ func (f findMessage) JSON() string {
 	return f.contentMessage.JSON()
 }
 
+// findMessageMatchCount holds JSON and string values for printing find --count command output.
+type findMessageMatchCount struct {
```
Comment (shtripat) [inline cmd/find.go:65]: I feel better name could be as below
```
type matchedFindCountStruct struct {
    TotalMatchedCount uint64 `json:"totalMatchedCount,omitempty"`
    TotalVersionsMatchedCount `json:"totalVersionsMatchedCount,omitmepty"`
}
```

**Comment by jiuker** [comment]
Indeed

**Pull request closed without merging**